src/Stan/Toml.hs

```diff
@@ -0,0 +1,98 @@
+{- |
+Copyright: (c) 2020 Kowainik
+SPDX-License-Identifier: MPL-2.0
+Maintainer: Kowainik <xrom.xkov@gmail.com>
+
+@tomland@ library integration. 'TomlCodec's for the 'Config' data type.
+-}
+
+module Stan.Toml
+    ( -- * Codecs
+      configCodec
+    ) where
+
+import Toml (Key, TomlCodec, (.=))
+
+import Stan.Category (Category (..))
+import Stan.Config (Check (..), CheckFilter (..), CheckScope (..), CheckType (..), Config (..))
+import Stan.Core.Id (Id (..))
+import Stan.Core.ModuleName (ModuleName (..))
+import Stan.Inspection (Inspection (..))
+import Stan.Observation (Observation (..))
+import Stan.Severity (Severity (..))
+
+import qualified Toml
+
+
+configCodec :: TomlCodec Config
+configCodec = Config
+    <$> Toml.list checkCodec "check" .= configChecks
+
+checkCodec :: TomlCodec Check
+checkCodec = Check
+    <$> checkTypeCodec .= checkType
+    <*> Toml.dioptional checkFilterCodec .= checkFilter
+    <*> Toml.dioptional checkScopeCodec  .= checkScope
+
+checkTypeCodec :: TomlCodec CheckType
+checkTypeCodec = Toml.enumBounded "type"
+
+----------------------------------------------------------------------------
+-- CheckFilter
+----------------------------------------------------------------------------
+
+checkInspection :: CheckFilter -> Maybe (Id Inspection)
+checkInspection = \case
+    CheckInspection idI -> Just idI
+    _ -> Nothing
```

Comment:
Clever naming conventing with naming matching functions same as constructors :brain: I haven't thought about that, and it actually looks neat :100: